history/src/block_stores/multiple_strategy_block_store.rs

```diff
@@ -0,0 +1,136 @@
+// A mixed block store,
+// Stores confirmed blocks in memory
+// Finalized blocks in long term storage of your choice
+// Fetches legacy blocks from faithful
+
+use crate::block_stores::inmemory_block_store::InmemoryBlockStore;
+use async_trait::async_trait;
+use solana_lite_rpc_core::{
+    structures::produced_block::ProducedBlock,
+    traits::block_storage_interface::{BlockStorageImpl, BlockStorageInterface},
+};
+use solana_rpc_client::nonblocking::rpc_client::RpcClient;
+use solana_rpc_client_api::config::RpcBlockConfig;
+use solana_sdk::{
+    commitment_config::{CommitmentConfig, CommitmentLevel},
+    slot_history::Slot,
+};
+use std::{
+    ops::Range,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+
+pub struct MultipleStrategyBlockStorage {
+    inmemory_for_storage: InmemoryBlockStore, // for confirmed blocks
+    persistent_block_storage: BlockStorageImpl, // for persistent block storage
+    faithful_rpc_client: Option<Arc<RpcClient>>, // to fetch legacy blocks from faithful
```

Comment:
that should go somewhere else.

Reply:
Nah we want to fall back to faithful if the link is provided..